contract_update_last_date_invoiced/views/contract_line.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2022 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record model="ir.ui.view" id="contract_line_form_view">
+        <field name="model">contract.line</field>
+        <field
+            name="name"
+        >contract.line.form (in contract_update_last_date_invoiced)</field>
+        <field name="inherit_id" ref="contract.contract_line_form_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//header" position="inside">
+                <button
+                    name="action_update_last_date_invoiced"
+                    type="object"
+                    string="Update Last Date Invoiced"
+                    invisible="not display_type"
```

Comment:
```suggestion
                    invisible="display_type"
```

The posible values for type are 
`    display_type = fields.Selection(
        selection=[("line_section", "Section"), ("line_note", "Note")],
        default=False,
        help="Technical field for UX purpose.",
    )`
so for "normal" lines the button is not shown..